google/cloud/forseti/common/gcp_api/groups_settings.py

```diff
@@ -0,0 +1,154 @@
+        """Constructor.
+
+        Args:
+            credentials (object): An google.auth credentials object. The group
+                settings API needs a service account credential with delegated
+                super admin role.
+            quota_max_calls (int): Allowed requests per <quota_period> for the
+                API.
+            quota_period (float): The time period to track requests over.
+            use_rate_limiter (bool): Set to false to disable the use of a rate
+                limiter for this service.
+        """
+        if not quota_max_calls:
+            use_rate_limiter = False
+
+        self._groups_settings = None
+
+        super(GroupsSettingsRepositoryClient, self).__init__(
+            API_NAME, versions=['v1'],
+            credentials=credentials,
+            quota_max_calls=quota_max_calls,
+            quota_period=quota_period,
+            use_rate_limiter=use_rate_limiter)
+
+    # Turn off docstrings for properties.
+    # pylint: disable=missing-return-doc, missing-return-type-doc
+    @property
+    def groups_settings(self):
+        """Returns a _GroupsSettingsRepository instance."""
+        if not self._groups_settings:
+            self._groups_settings = self._init_repository(
+                _GroupsSettingsRepository)
+        return self._groups_settings
+    # pylint: enable=missing-return-doc, missing-return-type-doc
+
+
+
+
+class _GroupsSettingsRepository(
+        repository_mixins.GetQueryMixin,
+        _base_repository.GCPRepository):
+    """Implementation of Groups Settings repository."""
+
+    def __init__(self, **kwargs):
+        """Constructor.
+
+        Args:
+            **kwargs (dict): The args to pass into GCPRepository.__init__()
+        """
+        super(_GroupsSettingsRepository, self).__init__(
+            key_field='groupUniqueId', component='groups', **kwargs)
+
+class GroupsSettingsClient(object):
+    """GSuite Groups Settings API Client."""
+
+    def __init__(self, global_configs, **kwargs):
+        """Initialize.
+
+        Args:
+            global_configs (dict): Global configurations.
+            **kwargs (dict): The kwargs.
+        """
+        credentials = api_helpers.get_delegated_credential(
+            global_configs.get('domain_super_admin_email'),
+            REQUIRED_SCOPES)
+
+        max_calls, quota_period = api_helpers.get_ratelimiter_config(
+            global_configs, API_NAME)
+
+        self.repository = GroupsSettingsRepositoryClient(
+            credentials=credentials,
+            quota_max_calls=max_calls,
+            quota_period=quota_period,
+            use_rate_limiter=kwargs.get('use_rate_limiter', True))
+
+    def get_groups_settings(self, group_id):
+        """Get the group settings for a given group.
+
+
+        https://developers.google.com/admin-sdk/groups-settings/v1/reference/groups/get
+
+        Args:
+            group_id (str): The gsuite group id to scope the request to.
```

Comment:
It would be helpful to add a clarifying detail that this is effectively the email address.
```
The gsuite group id to scope the request to, i.e. email address of the group.
```